subprojects/launcher/src/test/groovy/org/gradle/launcher/cli/WelcomeMessageActionTest.groovy

```diff
@@ -114,7 +114,7 @@ For more details see https://docs.gradle.org/42.0/release-notes.html''')
         given:
         def inputStreamWasClosed = false
         def inputStreamProvider = Mock(Function) {
-            apply(_) >> new ByteArrayInputStream("".bytes) {
+            apply(_) >> new ByteArrayInputStream(Byte.parseByte('0')) {
```

Comment:
What exactly goes wrong here in Groovy 4? The original code seems to work okay in groovysh.

Reply:
Weird. I don't understand what would cause that. It might be better to just use `(byte) 0` here though.

Reply:
What's _even stranger_ is that the BAIS ctor accepts a byte array, not a byte. Maybe something is wrong with spock's mocking?

Here's the short ST when I tried passing `new byte[0]`:

```
org.codehaus.groovy.runtime.typehandling.GroovyCastException: Cannot cast object '[B@5f8d9767' with class '[B' to class 'byte'
	at org.gradle.launcher.cli.WelcomeMessageActionTest.closes InputStream after reading features_closure3(WelcomeMessageActionTest.groovy:117)
	at groovy.lang.Closure.call(Closure.java:418)
	at org.spockframework.lang.SpecInternals.createMock(SpecInternals.java:53)
	at org.spockframework.lang.SpecInternals.createMockImpl(SpecInternals.java:305)
	at org.spockframework.lang.SpecInternals.createMockImpl(SpecInternals.java:295)
	at org.spockframework.lang.SpecInternals.MockImpl(SpecInternals.java:108)
	at org.gradle.launcher.cli.WelcomeMessageActionTest.closes InputStream after reading features(WelcomeMessageActionTest.groovy:116)
```